sdk/appconfiguration/app-configuration/src/models.ts

```diff
@@ -28,28 +30,35 @@ export interface ConfigurationSettingId {
 /**
  * Necessary fields for updating or creating a new configuration setting
  */
-export interface ConfigurationSettingParam extends ConfigurationSettingId {
+export type ConfigurationSettingParam<T = string> = ConfigurationSettingId & {
   /**
    * The content type of the setting's value
    */
   contentType?: string;
 
-  /**
-   * The setting's value
-   */
-  value?: string;
-
   /**
    * Tags for this key
    */
   tags?: { [propertyName: string]: string };
-}
+} & (T extends FeatureFlagValue | SecretReferenceValue
```

Comment:
could we do the conditional backwards? That is say if `T extends string` we make it optional, else it's required? That would make it easier to extend if we add more types in the future

Reply:
I was thinking it.
But I also was wondering if the service team comes up with a new setting kind that would accept undefined as value.

Reply:
Well, at that point I'll do `string | NewKind`.
Okay, will switch.

Reply:
Okay, I can add `string | NewKindValue` at that time.

However, upon trying - TS didn't allow me to do that. (Is that a bug?)

![image](https://user-images.githubusercontent.com/10452642/118198164-2eec3400-b405-11eb-82ed-b4b53bbe91a2.png)

[Playground link](https://www.typescriptlang.org/play?#code/FAUwHgDg9gTgLgAjgTwiBANAagQwDYCu6AvAgN7AJUJgIBcCAdgQLYBGIMA3MAL7ChIsRCjQIAkozicAZjgDGIAIwAeACoJSAZzgwAlowDmAPk1UN4aYwAmWhDv1HK1APzln1agDd8RFwzUeTyp+TwYKYO9fEAYHA0Mg4N4eQWh4JFR0SWkYOUUAJnUzOKNTUgQLMCtbTFxCEA8ENwjIhB96gMTPUOpwxs92v1jdeK7qZIFgeShGHTacfRw2PGUGbNkFZRVsaLLyQZiyMDolfIBmXgmpmbmfReWQfLWpDYLtuqI9sgO6I5Pzy5cBBAA)

Reply:
I'm maybe confused about what you're asking, it seems like the switch works okay? 

[simple playground](https://www.typescriptlang.org/play?#code/FAUwHgDg9gTgLgAjgTwiBANAagQwDYCu6AvAgN7AJUJgIBcCAdgQLYBGIMA3MAL7ChIsRCjQIAkozicAZjgDGIAIwAeSggAqCcNMYATAM4IDcGAEtGAcwQAfTLkIljpi5eAA+BKQpV5UKSBSGqggAPwMJuZWPLwIAGQIABRaOoGGzlHWoeQIAG74ROEZrgixDGR5BSAMGrwAlALAfowmleY4bHjKDJLSMHKKqthVnt75jnRkYHRKAEwAzLy8PM2t+e2dILM9Af0Kyl45y0A)

and if you had another nullable value (besides string) that seems doable too:

[nullable value example](https://www.typescriptlang.org/play?#code/FAUwHgDg9gTgLgAjgTwiBANAagQwDYCu6AvAgN7AIJgIBcCAdgQLYBGIMA3MAL7CiRYiFGgQA5Anjw5WeELkIlylBMnpM2HbnwHR4SVOgCSDOBwBmOAMYgAjAB4VAFQTgzDACYBnBF7gwASwYAcwQAH0wFInDxSWlZeXxo0j9AkOAAPgRSCiorKFMQUydDAH56VKDg7QQAMgQAChc3Iu9ffyqYiSkZOSj0UvIEADckkHKEFx46IdHFeiceAEp+YHyGPxGcQN67ehMzGEsbB2wxrJy5oloyMFpbACYAZh4ebnXN0Z2Eh-3Co+sdmyQzeawKn22AV2Tz+h2Odns3XifXOwLICDeQA)

Reply:
Ah.. I mistyped the Interface1. 
My bad, fixing it now.